catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/QueryResultIterable.java

```diff
@@ -90,6 +92,8 @@ public QueryResultIterable(CatalogFramework catalog, Function<Integer, QueryRequ
     class ResultQueryIterator implements Iterator<Result> {
```

Comment:
Doesn't quite seem like we can make it static as it relies on pageSize and other state fields. Thoughts on how to fix this or refactor to make it form a static class?